packages/sources/dns-query/test/integration/dnsProof.ts

```diff
@@ -0,0 +1,97 @@
+import { assertError, assertSuccess } from '@chainlink/ea-test-helpers'
+import type { AdapterRequest } from '@chainlink/types'
+import type { AddressInfo } from 'net'
+import request from 'supertest'
+import type { SuperTest, Test } from 'supertest'
+
+import type { SuiteContext } from './adapter.test'
+import {
+  mockDnsProofResponseError,
+  mockDnsProofResponseSuccessMalformed,
+  mockDnsProofResponseSuccess,
+} from './fixtures'
+
+export function dnsProofTests(context: SuiteContext): void {
+  let req: SuperTest<Test>
+  const id = '1'
+
+  beforeAll(() => {
+    req = request(`localhost:${(context.server.address() as AddressInfo).port}`)
+  })
+
+  describe('error calls', () => {
+    describe('when unsuccessfully requests the API', () => {
+      it('should throw and exception', async () => {
+        const data: AdapterRequest = {
+          id,
+          data: {
+            endpoint: 'dnsProof',
+            name: 'www5.infernos.io',
+            record: '0xf75519f611776c22275474151a04183665b7feDe',
+          },
+        }
+        mockDnsProofResponseError()
+
+        const response = await req
+          .post('/')
+          .send(data)
+          .set('Accept', '*/*')
+          .set('Content-Type', 'application/json')
+          .expect('Content-Type', /json/)
+          .expect(200)
+
+        assertError({ expected: 500, actual: response.body.providerStatusCode }, response.body, id)
```

Comment:
can we record the snapshots here as well?

Reply:
Hey Christopher, thanks for reviewing the code. As far as I'm aware, snapshots are being used for successful calls only. I just wanted to double check with you on that before I amend changes. Thank you.

Reply:
hey @1marcghannam sorry for the delayed response.  So we actually also record the snapshots for errors cases as well.  You can go ahead and amend this.